packages/react-flex/src/components/Flex/useFlexStyles.ts

```diff
@@ -16,11 +25,22 @@ export const useFlexStyles = (state: FlexState): FlexState => {
   const styles = useStyles();
   state.className = mergeClasses(styles.root, state.className);
 
-  // TODO: Hook up slot styles, for example:
-  // const exampleSlotClassName = useExampleSlotStyles(state);
-  // if (state.exampleSlot) {
-  //   state.exampleSlot.className = mergeClasses(exampleSlotClassName, state.exampleSlot.className);
-  // }
+  const direction = state.direction ?? 'row';
+  const horizontalAlign = state.horizontalAlign ?? 'normal';
+  const verticalAlign = state.verticalAlign ?? 'normal';
+
+  state.style = {
+    ...state.style,
+    display: state.inline ? 'inline-flex' : 'flex',
```

Comment:
inline styles are one of the slowest ways of applying styles. I can see why `grow`. `shrink` and `gap` are problematic, but I think the other properties can simply be classnames applied conditionally through `makeStyles`

Reply:
Also the inline styles will be very hard to override when used in the larger context of HVCs or applications .. (!important everywhere)

Reply:
Using class names would imply creating around 30 classes and we wouldn't be able to cover compound value scenarios that use `safe` and `unsafe`. If we want to cover that, then the number of classes almost doubles. Not only would this be very hard to maintain but also lengthy to implement. How much of a hit are talking about here?

Reply:
I also fail to see why overriding would be necessary if they can use the Flex props to change it.

Reply:
> I also fail to see why overriding would be necessary if they can use the Flex props to change it.

Looking at the base component it might not be obvious. But larger product teams might want to implement their own reusable components that other feature teams might want to tweak the style on. Unless prop drilling down to every flex is supported, it won't be possible to override those styles without using `!important` because they are inline style attributes

Reply:
> How much of a hit are talking about here?

Inserting inline styles can be x2 slower than inserting styles to `head`

Reply:
I think it's important to not complicate overriding styles but the performance concerns I think will depend a lot on the environment. Opting for the className alternative will also expand the bundle size so it's a tradeoff we need to consider.
As a compromise, what do you think about using CSS vars for all props so that bundle size remains marginally the same, overriding styles is no longer an issue and then we analyze the performance for the difference use cases?

Reply:
hmm, idea having everything handled by css vars sounds indeed interesting to me!

No idea what reflow/re-paint consequences it might have though.

Reply:
`Flex` is one of the most commonly used components in large applications (i.e. Teams), even when it's not really needed, you're still inserting inline styles on every render which is a lot slower compared to CSS classes. I can see why you'd need this for styles with numerical values, but you save a lot of runtime cost by using `makeStyles` for the other combinations, especially since we'll be using a build time plugin to remove the runtime cost of generating those classnames

Reply:
> you're still inserting inline styles on every render which is a lot slower compared to CSS classes

in this case we'll re-insert only css variables, not direct styles (which is slow in general). I haven't seen any benchmarks for that, do you have any (- I'll need to check css vars spec how it affects layout/paint) ?

Reply:
in RFC #17815 I made benchmarks for inline style insertion of CSS vars compared to consuming a classname, the result of > 2 times slower. However, this could be overkill since there are much more CSS variables used in that benchmark, but I can run a benchmark pretty easily for this scenario if needed, just let me know

Reply:
Also I am OK using CSS variables here if you want, just pointing out that the insertion will be slower.

Reply:
if you can run those benchmarks just for css vars inline re-insertion that would be great! thx